library/std/src/collections/hash/set.rs

```diff
@@ -12,13 +12,6 @@ use crate::ops::{BitAnd, BitOr, BitXor, Sub};
 
 use super::map::{map_try_reserve_error, RandomState};
 
-// Future Optimization (FIXME!)
-// ============================
-//
-// Iteration over zero sized values is a noop. There is no need
-// for `bucket.val` in the case of HashSet. I suppose we would need HKT
-// to get rid of it properly.
-
```

Comment:
This was [added](https://github.com/rust-lang/rust/commit/ae7342a56a24eac539e3d4b13cd49c6719908426#diff-743ffa68e30e5a2fdfcb42c0f2858887d55c36f300a9187ea011877c671a85ebR28-R33) way before hashbrown. Now this doesn't make any sense.